app/jobs/operation_job.rb

```diff
@@ -3,53 +3,58 @@ class OperationJob < ApplicationJob
 
   rescue_from StandardError, with: :operation_raised_error
 
-  before_perform { |job| job.operation_started }
-
-  after_perform { |job| job.operation_completed }
+  before_perform :operation_started, if: :operation
```

Comment:
Please put a comment explaining why we need to check for operation being present instead of just being able to assume it. I believe it's because the Operation might have gotten deleted in the other process possibly?